iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateCouplingDevice.java

```diff
@@ -0,0 +1,125 @@
+        BusbarSection bbs1 = network.getBusbarSection(bbsId1);
+        if (bbs1 == null) {
+            bbsDoesNotExist(bbsId1, reporter, throwException);
+            return;
+        }
+
+        BusbarSection bbs2 = network.getBusbarSection(bbsId2);
+        if (bbs2 == null) {
+            bbsDoesNotExist(bbsId2, reporter, throwException);
+            return;
+        }
+
+        VoltageLevel voltageLevel1 = bbs1.getTerminal().getVoltageLevel();
+        VoltageLevel voltageLevel2 = bbs2.getTerminal().getVoltageLevel();
+        if (voltageLevel1 != voltageLevel2) {
+            LOGGER.error("Busbar sections {} and {} are in two different voltage levels.", bbsId1, bbsId2);
+            busbarsInDifferentVoltageLevels(reporter, bbsId1, bbsId2);
+            if (throwException) {
+                throw new PowsyblException(String.format("Busbar sections %s and %s are in two different voltage levels.", bbsId1, bbsId2));
+            }
+            return;
+        }
+        int breakerNode1 = voltageLevel1.getNodeBreakerView().getMaximumNodeIndex() + 1;
+        int breakerNode2 = breakerNode1 + 1;
+        int bbs1Node = bbs1.getTerminal().getNodeBreakerView().getNode();
+        int bbs2Node = bbs2.getTerminal().getNodeBreakerView().getNode();
+
+        createNBBreaker(breakerNode1, breakerNode2, "", "NEW", voltageLevel1.getNodeBreakerView(), false);
+        createNBDisconnector(bbs1Node, breakerNode1, String.valueOf(bbs1Node), "NEW", voltageLevel1.getNodeBreakerView(), false);
+        createNBDisconnector(bbs2Node, breakerNode2, String.valueOf(bbs2Node), "NEW", voltageLevel1.getNodeBreakerView(), false);
+
+        BusbarSectionPosition position1 = bbs1.getExtension(BusbarSectionPosition.class);
+        BusbarSectionPosition position2 = bbs2.getExtension(BusbarSectionPosition.class);
+        int nbOpenDisconnectors = 0;
+        if (position1 != null) {
+            if (position2 != null) {
+                List<BusbarSection> bbsList1 = voltageLevel1.getNodeBreakerView().getBusbarSectionStream()
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class) != null)
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class).getSectionIndex() == position1.getSectionIndex())
+                        .filter(b -> !b.getId().equals(bbsId1)).collect(Collectors.toList());
+                List<BusbarSection> bbsList2 = voltageLevel2.getNodeBreakerView().getBusbarSectionStream()
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class) != null)
+                        .filter(b -> b.getExtension(BusbarSectionPosition.class).getSectionIndex() == position2.getSectionIndex())
+                        .filter(b -> !b.getId().equals(bbsId2)).collect(Collectors.toList());
+                if (!(bbsList1.size() == 1 && position1.getSectionIndex() == position2.getSectionIndex())) {
```

Comment:
Easier to read  directly: `if (bbsLis1.size() != 1 || position1.getSectionIndex() != position2.getSectionIndex())`

Also add a comment here to explain why you do this check.